File: WebDriverTests_Java/WebDriverTests/src/test/java/core/ConstVariablesTestBase.java
package core;

import core.api.Api;
import org.codehaus.jackson.annotate.JsonAutoDetect;
import org.codehaus.jackson.annotate.JsonMethod;
import org.codehaus.jackson.map.DeserializationConfig;
import org.codehaus.jackson.map.ObjectMapper;

public class ConstVariablesTestBase
{
    public String getApplicationBaseUrl()
    {
//	return "dev.ddelivery.ru/";
        return "stage.ddelivery.ru/";
    }

    public String adminName = "v.e@ddelivery.ru";
    public String adminPass = "LbREWCnphA";
    public String adminKey = "9c205350bc5f193bc100a55bac2109f0";

    public String legalEntityName = "test_legalEntity";
    public String legalUserName = "test_legalUser";
    public String legalPickupName = "test_legalPickup";
    public String companyName = "test_via";
    public String companyPickupName = "test_Pickup";
    public String deliveryPointName = "test_deliverypoint";

    public String userNameAndPass = "tester@user.ru";
    public String pickupNameAndPass = "tester@pickup.ru";
    public String userWarehouseName = "test_userWarehouses_via";
    public String userShopName = "test_userShops_via";

    public String weightName = "test_via_Weight";
    public double weightMin = 2;
    public double weightMax = 16;

    public String sideName = "test_via_Side";
    public double side1Min = 1;
    public double side2Min = 2;
    public double side3Min = 3;
    public double side1Max = 40;
    public double side2Max = 50;
    public double side3Max = 60;

    public int marginsPickup = 11;
    public int marginsCourirs = 12;
    public int marginsSelf = 13;

    public String deliveryPointAddress = "Ленинский проспект 127";
    public String deliveryPointLongitude = "37.477079";
    public String deliveryPointLatitude = "55.645873";

    public String deliveryPointAddress2 = "ул. Салова, 27Литер АД, пом. 35";
    public String deliveryPointLongitude2 = "30.372815";
    public String deliveryPointLatitude2 = "59.895735";

    public Api apiRequest = new Api(getApplicationBaseUrl());
    public ObjectMapper mapper = new ObjectMapper().setVisibility(JsonMethod.FIELD, JsonAutoDetect.Visibility.ANY).
            configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.WebPages/ConstVariablesTestBase.cs
using System;
using Autotests.Utilities.ApiTestCore;
using Autotests.WebPages.Pages;

namespace Autotests.WebPages
{
    public class ConstVariablesTestBase : SimpleFunctionalTestBase
    {
// 		public override string ApplicationBaseUrl { get { return "dev:0DShabby7maiden&0Edata@dev.ddelivery.ru"; } }
        public override string ApplicationBaseUrl { get { return "stage.ddelivery.ru"; } }

//        public override string ApplicationBaseUrl { get { return "cabinet.ddelivery.ru"; } }

        public override void SetUp()
        {
            base.SetUp();
            
            adminName = "v.e@ddelivery.ru";
            adminPass = "LbREWCnphA";
            adminKey = "9c205350bc5f193bc100a55bac2109f0";

            legalEntityName = "test_legalEntity";
            legalUserName = "test_legalUser";
            legalPickupName = "test_legalPickup";
            companyName = "test_via";
            companyPickupName = "test_Pickup";
            deliveryPointName = "test_deliverypoint";

            userNameAndPass = "tester@user.ru";
            pickupNameAndPass = "tester@pickup.ru";
            userWarehouseName = "test_userWarehouses_via";
            userShopName = "test_userShops_via";

            weightName = "test_via_Weight";
            weightMin = 2;
            weightMax = 16;

            sideName = "test_via_Side";
            side1Min = 1;
            side2Min = 2;
            side3Min = 3;
            side1Max = 40;
            side2Max = 50;
            side3Max = 60;

            marginsPickup = 11;
            marginsCourirs = 12;
            marginsSelf = 13;

            deliveryPointAddress = "Ленинский проспект 127";
            deliveryPointLongitude = "37.477079";
            deliveryPointLatitude = "55.645873";
            
            deliveryPointAddress2 = "ул. Салова, 27Литер АД, пом. 35";
            deliveryPointLongitude2 = "30.372815";
            deliveryPointLatitude2 = "59.895735";

            apiRequest = new Api(ApplicationBaseUrl);
            TimeZoneInfo moscowTimeZone = TimeZoneInfo.FindSystemTimeZoneById("Russian Standard Time");
            DateTime moscowDateTime = TimeZoneInfo.ConvertTimeFromUtc(DateTime.UtcNow, moscowTimeZone);
            nowDate = moscowDateTime;
        }

        public string adminName;
        public string adminPass;
        public string adminKey;

        public string userNameAndPass;
        public string pickupNameAndPass;
        public string userWarehouseName;
        public string userShopName;

        public string legalEntityName;
        public string legalUserName;
        public string legalPickupName;
        public string companyName;
        public string companyPickupName;
        public string deliveryPointName;

        public string weightName;
        public double weightMin;
        public double weightMax;

        public string sideName;
        public double side1Min;
        public double side2Min;
        public double side3Min;
        public double side1Max;
        public double side2Max;
        public double side3Max;

        public string deliveryPointAddress;
        public string deliveryPointLongitude;
        public string deliveryPointLatitude;
        public string deliveryPointAddress2;
        public string deliveryPointLongitude2;
        public string deliveryPointLatitude2;

        public int marginsPickup;
        public int marginsCourirs;
        public int marginsSelf;

        public Api apiRequest;
        public DateTime nowDate;

        public string GetOdrerIdTakeOutUrl()
        {
            var page = new LoginPage();
            return page.GetUrl()
                .Replace("http://", "")
                .Replace(".ddelivery.ru/user/orders/view/", "")
                .Replace("stage", "")
                .Replace("dev", "");
       }
    }
}